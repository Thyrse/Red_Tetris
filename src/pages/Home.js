/* eslint-disable array-callback-return */
// REACT
import React, { useState, useEffect, useContext, useCallback } from "react";
// REACT ROUTER
import { Link, useHistory } from "react-router-dom";
import arrowRight from "../img/arrow_right.png";
import arrowRightWhite from "../img/arrow_right_white.png";
import Chat from "../components/Chat";
import socketIOClient from "socket.io-client";
// import { yoloEmit } from "../../public/yolo";

/**
 * Component that displays the patient page,
 * contains the list of patients,
 * as well as the components for display and editing,
 * drawers and modal
 */
const Home = () => {
  const socket = socketIOClient.connect("http://localhost:4000");
  useEffect(() => {
    console.log("Passing here ==>", socket);
    socket.emit("yolo", "Salut les michtos");
  });
  return (
    <>
      <div className="container-fluid">
        <div className="row">
          <div className="col-12 col-lg-6">
            <div className="room-container pb-4 m-3 shadow">
              <div className="bg-head chat-container__box-title row align-items-center justify-content-center border-bottom p-3">
                <h3>Rooms</h3>
              </div>
              <div className="p-3">
                <form action="">
                  <div className="room-container__create">
                    <input
                      type="text"
                      name="username"
                      id="username"
                      className="form-control"
                      placeholder="Enter room name..."
                    />
                    <button className="btn btn-img btn-outline-arrow">
                      <img src={arrowRightWhite} alt="Validation button" />
                    </button>
                  </div>
                </form>
              </div>
              <div className="container-fluid room-container__list">
                <div className="row text-center room-item">
                  <div className="col-8">
                    <span>Room name</span>
                  </div>
                  <div className="room-join col-2">
                    <span>3 / 4</span>
                  </div>
                  <div className="col-2">
                    <button className="btn btn-img">
                      <img src={arrowRightWhite} alt="Validation button" />
                    </button>
                  </div>
                </div>
                <div className="row text-center room-item">
                  <div className="col-8">
                    <span>Room name</span>
                  </div>
                  <div className="room-join col-2">
                    <span>3 / 4</span>
                  </div>
                  <div className="col-2">
                    <button className="btn btn-img">
                      <img src={arrowRightWhite} alt="Validation button" />
                    </button>
                  </div>
                </div>
                <div className="row text-center room-item">
                  <div className="col-8">
                    <span>Room name</span>
                  </div>
                  <div className="room-join col-2">
                    <span>3 / 4</span>
                  </div>
                  <div className="col-2">
                    <button className="btn btn-img">
                      <img src={arrowRightWhite} alt="Validation button" />
                    </button>
                  </div>
                </div>
                <div className="row text-center room-item">
                  <div className="col-8">
                    <span>Room name</span>
                  </div>
                  <div className="room-join col-2">
                    <span>3 / 4</span>
                  </div>
                  <div className="col-2">
                    <button className="btn btn-img">
                      <img src={arrowRightWhite} alt="Validation button" />
                    </button>
                  </div>
                </div>
                {/* <div className="room-item">
                  <div>
                    <span>Room name</span>
                  </div>
                  <div className="room-join">
                    <span>3 / 4</span>
                  </div>
                  <div>
                    <button className="btn btn-img btn-outline-arrow">
                      <img src={arrowRight} alt="Validation button" />
                    </button>
                  </div>
                </div> */}
              </div>
            </div>
          </div>
          <Chat />
        </div>
      </div>
    </>
  );
};

export default Home;
