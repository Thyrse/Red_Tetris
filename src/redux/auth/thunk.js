import { setUserData, setDisconnectUser, setUserDataError } from "./actions";

// THUNK

export const login = (user) => async (dispatch) => {
  console.log("USER SENT TO THUNK ==>", user);
  //   try {
  //     return auth;
  //   } catch (err) {
  //     console.error("AUTHENTICATE_ERROR", err);
  //     dispatch(setUserDataError(err.toString()));
  //   }
};
