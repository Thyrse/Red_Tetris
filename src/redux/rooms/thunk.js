export const hydrateStore = async (dispatch) => {
  // console.log("HYDRATESTORE");
  try {
    // fill store
    await Promise.all([
      dispatch(fetchWorkspaces()),
      dispatch(fetchChatrooms()),
    ]);
  } catch (e) {
    console.error("HYDRATESTORE_ERROR", e);
  }
};
