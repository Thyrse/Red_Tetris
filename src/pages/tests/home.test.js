import React from "react";
import { mount, shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Home from "../Home";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import { Provider } from "react-redux";
import renderer from "react-test-renderer";
import socketIOClient from "socket.io-client";
import { string } from "yargs";

const socket = socketIOClient.connect("http://localhost:4000");

const middlewares = [thunk];

const mockJest = jest.fn();

const mockStore = configureStore(middlewares);
const initialState = {
  userData: {
    userDatas: {
      username: "Ziphlot",
      socketID: socket.id,
      inGame: false,
      ownedRooms: [],
      room: "Lobby",
    },
  },
  listUsers: {
    usersList: [
      {
        id: "1d2d3d",
        username: "Ziphlot",
        inGame: false,
        room: "Lobby",
      },
    ],
  },
  roomsList: {
    roomsList: [
      {
        id: "1d2d3d",
        name: "salut",
        owner: "Thyrse",
        members: [],
        size: 5,
      },
    ],
  },
};

const store = mockStore(initialState);

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useHistory: () => ({
    push: mockJest,
  }),
}));

const props = {
  socket: socket,
};

it("Homepage rendering upon arrival", () => {
  const wrapper = renderer
    .create(
      <Provider store={store}>
        <Home {...props} />
      </Provider>
    )
    .toJSON();
  expect(wrapper).toMatchSnapshot();
});

it("Should validate the room creation if name valid", () => {
  const fakeEvent = { preventDefault: () => console.log("preventDefault") };
  const wrapper = mount(
    <Provider store={store}>
      <Home {...props} />
    </Provider>
  );
  const roomName = "Salut";
  const event = { target: { value: roomName } };
  expect(wrapper.find(".control-room").simulate("change", event));
  wrapper.find(".form-room").simulate("submit", fakeEvent);
  expect(wrapper.find(".control-room").props().value).toBe("");
});

// it("Should redirect on form submission if an username is provided", () => {
//   const fakeEvent = { preventDefault: () => console.log("preventDefault") };
//   const wrapper = mount(
//     <Provider store={store}>
//       <Home {...props} />
//     </Provider>
//   );
//   const username = "Ziphlot";
//   const event = { target: { value: username } };
//   expect(wrapper.find(".form-control").simulate("change", event));
//   wrapper.find(".form-login").simulate("submit", fakeEvent);
//   expect(mockJest).toHaveBeenCalledWith("/home");
// });
