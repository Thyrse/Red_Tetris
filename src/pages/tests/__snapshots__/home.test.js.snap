// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage rendering upon arrival 1`] = `
<div
  className="container-fluid"
>
  <div
    className="row"
  >
    <div
      className="col-12 col-lg-6"
    >
      <div
        className="room-container pb-4 m-3 shadow"
      >
        <div
          className="bg-head chat-container__box-title row align-items-center justify-content-center border-bottom p-3"
        >
          <h3>
            Rooms
          </h3>
        </div>
        <div
          className="p-3"
        >
          <form
            className="form-room"
            onSubmit={[Function]}
          >
            <div
              className="room-container__create"
            >
              <input
                className="form-control control-room"
                id="roomName"
                name="roomName"
                onChange={[Function]}
                placeholder="Enter room name..."
                type="text"
                value=""
              />
              <span
                aria-describedby={null}
                className="p-1"
                onBlur={[Function]}
                onFocus={[Function]}
                onMouseLeave={[Function]}
                onMouseOver={[Function]}
                onTouchEnd={[Function]}
                onTouchStart={[Function]}
                title="Room must contain between 1 and 10 alphanumeric characters."
              >
                <button
                  className="btn btn-img btn-outline-arrow"
                  disabled={true}
                >
                  <img
                    alt="Validation button"
                    src={Object {}}
                  />
                </button>
              </span>
            </div>
          </form>
        </div>
        <div
          className="container-fluid room-container__list"
        >
          <div
            className="row text-center room-item"
          >
            <div
              className="col-8"
            >
              <span>
                Salut
              </span>
            </div>
            <div
              className="room-join col-2"
            >
              <span>
                0
                /
                5
              </span>
            </div>
            <div
              className="room-button col-2"
            >
              <button
                className="btn btn-img btn-join-room"
                onClick={[Function]}
              >
                <img
                  alt="Validation button"
                  src={Object {}}
                />
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="col-12 col-lg-6 chat-container"
    >
      <div
        className="container-fluid row mx-auto d-flex justify-content-center m-3"
      >
        <div
          className="col-12 border-dark rounded chat-container__box shadow"
        >
          <div
            className="bg-head chat-container__box-title row align-items-center justify-content-center border-bottom p-3"
          >
            <h3>
              Messagerie
            </h3>
          </div>
          <div
            className="row"
          >
            <div
              className="col-9 pt-2 chat-messages"
              id="zone_chat"
            >
              <div
                className="row"
              />
            </div>
            <div
              className="col-3 chat-users shadow__light"
            >
              <ul>
                <li>
                  ZIPHLOT
                </li>
              </ul>
            </div>
          </div>
          <form
            action=""
            id="formulaire_chat"
            onSubmit={[Function]}
          >
            <div
              className="row msg-foot border-top"
            >
              <div
                className="col-md-9 col-lg-10 msg-new secondary-font"
              >
                <input
                  autoFocus={true}
                  className="msg-new-input"
                  id="message"
                  name="message"
                  onChange={[Function]}
                  placeholder="Message..."
                  size="50"
                  spellCheck="false"
                  type="text"
                  value=""
                />
                <input
                  name="roomie"
                  type="hidden"
                />
              </div>
              <div
                className="col-md-3 col-lg-2 text-right"
              >
                <input
                  name="profile_id"
                  type="hidden"
                  value=""
                />
                <button
                  className="btn btn-white msg-button"
                  disabled={true}
                  id="envoi_message"
                  type="submit"
                >
                  Send â‡§
                </button>
              </div>
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;
